### PR TITLE
Added stake to register validator

### PR DESCRIPTION
**Pull request by dusannosovic-ethernal**
# Description

This PR includes support for staking alongside validators registration

# Changes include

- [ ] Bugfix (non-breaking change that solves an issue)
- [ ] Hotfix (change that solves an urgent issue, and requires immediate attention)
- [X] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

# Breaking changes

Please complete this section if any breaking changes have been made, otherwise delete it

# Checklist

- [X] I have assigned this PR to myself
- [X] I have added at least 1 reviewer
- [ ] I have added the relevant labels
- [ ] I have updated the official documentation
- [ ] I have added sufficient documentation in code

## Testing

- [ ] I have tested this code with the official test suite
- [ ] I have tested this code manually

### CODE REVIEW

File: e2e-polybft/e2e/consensus_test.go [hunk 1]
```diff
@@ -238,14 +238,11 @@ func TestE2E_Consensus_RegisterValidator(t *testing.T) {
 	require.NoError(t, firstValidator.RegisterValidator())
```
Comment (goran-ethernal) [inline e2e-polybft/e2e/consensus_test.go]: Change this to also use `RegisterValidatorWithStakeFunction`.

**Review by goran-ethernal — COMMENTED** [review]
Change `type registerResult struct` to this, so it receives the `Amount` that was staked:

```
type registerResult struct {
	ValidatorAddress string `json:"validatorAddress"`
	KoskSignature    string `json:"koskSignature"`
	Amount           *big.Int `json:"amount"`
}
```
and modify `GetOutput` function accordingly to show staked amount in output.

File: command/validator/registration/params.go [hunk 2]
```diff
@@ -35,8 +44,9 @@ func (rr registerResult) GetOutput() string {
 	buffer.WriteString("\n[VALIDATOR REGISTRATION]\n")
 
 	vals := make([]string, 0, 2)
-	vals = append(vals, fmt.Sprintf("Validator Address|%s", rr.validatorAddress))
-	vals = append(vals, fmt.Sprintf("KOSK Signature|%s", rr.koskSignature))
+	vals = append(vals, fmt.Sprintf("Validator Address|%s", rr.ValidatorAddress))
+	vals = append(vals, fmt.Sprintf("KOSK Signature|%s", rr.KoskSignature))
+	vals = append(vals, fmt.Sprintf("Amount|%s", rr.Amount))
```
Comment (goran-ethernal) [inline command/validator/registration/params.go]:
```suggestion
	vals = append(vals, fmt.Sprintf("Amount Staked|%s", rr.Amount))
```

File: e2e-polybft/e2e/consensus_test.go (hunk 1 above)
Comment (goran-ethernal) [reply e2e-polybft/e2e/consensus_test.go]: This was not changed as per comment. You still have `firstValidator.RegisterValidator`, and then `firstValidator.Stake`. Merge them into one call with `firstValidator.RegisterValidatorWithStake(stakeAmount)`.

**Review by goran-ethernal — DISMISSED** [review]

File: command/helper/helper.go [hunk 3]
```diff
@@ -251,6 +251,15 @@ func ParseAmount(amount string) (*big.Int, error) {
 	return result, nil
 }
 
+func ParseAmountAllowZero(amount string) (*big.Int, error) {
+	result, ok := new(big.Int).SetString(amount, 0)
+	if !ok || result.Cmp(big.NewInt(0)) < 0 {
+		return nil, fmt.Errorf("amount %s should be numerical value greater than zero", amount)
+	}
```
Comment (Stefan-Ethernal) [inline command/helper/helper.go]:
```suggestion
	if !ok {
		return nil, fmt.Errorf("failed to convert provided value into a number (%s)", amount)
	}
	
	if result.Cmp(big.NewInt(0)) < 0 {
	        return nil, fmt.Errorf("provided value (%d) is less than zero", result)
	}
```

**Review by Stefan-Ethernal — COMMENTED** [review]
The `TestE2E_Consensus_RegisterValidator` test is failing at the moment.

File: command/validator/registration/params.go [hunk 4]
```diff
@@ -35,8 +44,9 @@ func (rr registerResult) GetOutput() string {
 	buffer.WriteString("\n[VALIDATOR REGISTRATION]\n")
 
 	vals := make([]string, 0, 2)
```
Comment (goran-ethernal) [inline command/validator/registration/params.go]:
```suggestion
	vals := make([]string, 0, 3)
```

File: command/validator/registration/register_validator.go [hunk 5]
```diff
@@ -79,6 +93,23 @@ func runCommand(cmd *cobra.Command, _ []string) error {
...
+		if err != nil {
+			return err
+		}
+
+		if receipt.Status == uint64(types.ReceiptFailed) {
+			return fmt.Errorf("approve transaction failed on block %d", receipt.BlockNumber)
+		}
+	}
+
 	newValidatorAccount, err := wallet.NewAccountFromSecret(secretsManager)
```
Comment (goran-ethernal) [inline command/validator/registration/register_validator.go]: Do we need this if we created: `validatorAccount, err := validatorHelper.GetAccount(params.accountDir, params.accountConfig)`?

**Review by Stefan-Ethernal — APPROVED** [review]

**Review by goran-ethernal — APPROVED** [review]

**Pull request merged**